go/updater/updater.go

```diff
@@ -335,6 +335,26 @@ func (u *Updater) unpack(filename string) (string, error) {
 	return unzipDestination, nil
 }
 
+func (u *Updater) ensureAssetExists(asset keybase1.Asset) error {
+	if !strings.HasPrefix(asset.Url, "http://") && !strings.HasPrefix(asset.Url, "https://") {
+		u.log.Debug("Skipping re-check for non-http asset")
+		return nil
+	}
+	u.log.Debug("Checking asset still exists") // In case the update was revoked
+	resp, err := http.Head(asset.Url)
+	if err != nil {
+		return err
+	}
+	if resp == nil {
+		return fmt.Errorf("No response")
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Update is no longer available (%d)", resp.StatusCode)
+	}
+	defer libkb.DiscardAndCloseBody(resp)
```

Comment:
I think just to be safe... I'll move the defer before the status code check (in which case it does need to be a differ).. thx!